sycl/include/CL/sycl/detail/pi.h

```diff
@@ -50,11 +50,36 @@
 #define _PI_H_VERSION_MAJOR 8
 #define _PI_H_VERSION_MINOR 10
 
+// Versions of plugins
+#define _PI_CUDA_PLUGIN_VERSION 1
+#define _PI_LEVEL_ZERO_PLUGIN_VERSION 1
+#define _PI_HIP_PLUGIN_VERSION 1
+#define _PI_ESIMD_PLUGIN_VERSION 1
+#define _PI_OPENCL_PLUGIN_VERSION 1
```

Comment:
Is there a reason to have it in pi.h rather than in the individual plugins? Apart from a handful of backend-specific enums, pi.h is backend-agnostic so if the plugin version is used by the individual plugins only it would be nicer to have them maintain them locally.

Reply:
Plugin versions should not be exposed like this. `piPluginInit` should fill this info for the user to read.

Reply:
@smaslov-intel is right. The plugins are separate libraries so as long as the runtime library and the plugins use the same major version of the PI API you should be able to use different versions of the plugin libraries. As such, having the plugin version in definitions in the headers may be misleading as the used plugin library may not actually be that version.

Reply:
Yes that is fair enough. I don't immediately see how the user could query the Plugin version through the pi_plugin struct. Would an environment variable be more user friendly? Something like `SYCL_PI_PRINT_PLUGIN_VERSION` perhaps could print the active plugin version to screen.

Reply:
I think you're right that there is no current way for the user to query the version, though the information is available to the runtime through `_pi_plugin.PluginVersion`. You could make a case for `SYCL_PI_TRACE` at a certain level should report the plugin version after `piPluginInit` is called.

Reply:
What is the purpose of those prints? Who is expected to use that info and how?
SYCL_PI_TRACE=1 already prints  info about loaded plugins, so it could be extended to print basic information about the plugins loaded.

Reply:
SYCL_PI_TRACE=1 modified to print PluginVersion.